design-proposals/multus-annotation-customization.md

```diff
@@ -0,0 +1,59 @@
+# Overview
+Providing a way to customize the Multus pod annotation generated by the `virt-controller` where only a subset of the 
+Multus contract is currently mapped:
+- Name
+- Namespace
+- Interface name
+- MAC Address (if set in `interface` definition)
+
+## Motivation
+Improving the Multus support. As an example, customizing the IP through the annotation avoids having inconsistency between
+the IP dynamically assigned by Multus daemonset to the pod and the one defined guest level.
+
+## Goals
+- Consistency and visibility by having pod level what is defined in the guest
+- [Multus service](https://github.com/k8snetworkplumbingwg/multus-service) could work as expected by 
+having the Multus endpoint slice targeting a pod IP matching what's in the VM/guest 
+- Any Multus configuration can be customized, to see all the fields customizable, you can look at [NetworkSelectionElement](https://github.com/k8snetworkplumbingwg/multus-cni/blob/acfbd42719545844248ecc35cc93d0ce367f87e5/pkg/types/types.go#L128)
+
+## Non Goals
+- Altering the current behavior, the current state covers 90% of use cases. Only enhancing the Multus support by 
+providing a way to customize it, if needed, without needing KubeVirt maintainers to actively maintain it
+
+# Design
+1. End user annotates VMI with `k8s.v1.cni.cncf.io/networks-customization`, same contract as 
+`k8s.v1.cni.cncf.io/networks`
+2. Pod `virt-launcher` inherits from all the VMI annotations (an already existing behavior)
+3. A new mutating webhook is observing `virt-launcher` pod creation and updates, and merges `k8s.v1.cni.cncf.io/networks-customization` into
+`k8s.v1.cni.cncf.io/networks` based on the network's `name` and `namespace`
+
+**NOTE:** Existing Multus fields managed by the `virt-controller` are immutable. Any change to `interface` or `mac` is
+rejected by the webhook
```

Comment:
From a maintenance point of view, this is a burden to track. While today these are reserved, in the future others may be used by the core.
As such, I think it will be better to only warn in the documentation about this but not restrict/validate in-code.